src/routing_node.rs

```diff
@@ -232,6 +233,33 @@ impl RoutingNode {
           self.set_cache_options(preserve_cache_options);
     }
 
+    fn handle_on_connect(&mut self, connection: ::crust::Connection) {
+        match self.core.state() {
+            &::routing_core::State::Disconnected => {
+                self.core.add_bootstrap_connection(connection.clone());
+                ignore(self.send_hello(connection, None));
+                return;
+            },
+            &::routing_core::State::Bootstrapped => {
+                self.crust_service.drop_node(connection);
```

Comment:
combined with above comment, here we would also send a hello (which will indicate that we are a client connecting)